react/src/components/BoxReconciliationOverlay/BoxReconciliationOverlay.tsx

```diff
@@ -0,0 +1,233 @@
+import { useCallback, useContext, useEffect, useMemo } from "react";
+import { useMutation, useQuery, useReactiveVar } from "@apollo/client";
+import {
+  boxReconciliationLocationFormDataVar,
+  boxReconciliationProductFormDataVar,
+  boxReconciliationOverlayVar,
+} from "queries/cache";
+import { useErrorHandling } from "hooks/useErrorHandling";
+import { useNotification } from "hooks/useNotification";
+import {
+  BoxState,
+  ShipmentByIdWithProductsAndLocationsQuery,
+  ShipmentByIdWithProductsAndLocationsQueryVariables,
+  ShipmentDetail,
+  UpdateShipmentWhenReceivingMutation,
+  UpdateShipmentWhenReceivingMutationVariables,
+} from "types/generated/graphql";
+import { GlobalPreferencesContext } from "providers/GlobalPreferencesProvider";
+import { SHIPMENT_BY_ID_WITH_PRODUCTS_AND_LOCATIONS_QUERY } from "queries/queries";
+import { UPDATE_SHIPMENT_WHEN_RECEIVING } from "queries/mutations";
+import {
+  BoxReconciliationView,
+  ILocationData,
+  IProductWithSizeRangeData,
+} from "./components/BoxReconciliationView";
+
+export interface IBoxReconciliationOverlayData {
+  shipmentDetail: ShipmentDetail;
+}
+
+export function BoxReconciliationOverlay() {
+  const { createToast } = useNotification();
+  const { triggerError } = useErrorHandling();
+  const { globalPreferences } = useContext(GlobalPreferencesContext);
+  const baseId = globalPreferences.selectedBaseId;
+  const boxReconciliationOverlayState = useReactiveVar(boxReconciliationOverlayVar);
+  const boxReconciliationProductFormState = useReactiveVar(boxReconciliationProductFormDataVar);
+  const boxReconciliationLocationFormState = useReactiveVar(boxReconciliationLocationFormDataVar);
+
+  const onOverlayClose = useCallback(() => {
+    boxReconciliationOverlayVar({
+      isOpen: false,
+      labelIdentifier: undefined,
+      shipmentId: undefined,
+    });
+
+    boxReconciliationLocationFormDataVar({
+      locationId: undefined,
+    });
+
+    boxReconciliationProductFormDataVar({
+      productId: undefined,
+      sizeId: undefined,
+      numberOfItems: undefined,
+    });
+  }, []);
+
+  const { loading, error, data } = useQuery<
+    ShipmentByIdWithProductsAndLocationsQuery,
+    ShipmentByIdWithProductsAndLocationsQueryVariables
+  >(SHIPMENT_BY_ID_WITH_PRODUCTS_AND_LOCATIONS_QUERY, {
+    variables: {
+      shipmentId: boxReconciliationOverlayState.shipmentId || "",
+      baseId: baseId || "",
+    },
+    skip: !boxReconciliationOverlayState.shipmentId || !baseId,
```

Comment:
@vahidbazzaz why are you checking for this? Is this a possible scenario that the BoxReconciliation is called without shipmentId or baseId?
If so, we should probably do some error handling for this case. It should not be possible to call the overlay without the two values

Reply:
@HaGuesto yes indeed, as I used a reactive variable as soon as the page overly renders it will trigger the query. I used `skip` to only run the query when both `shipmentId` and 'baseId' are set. But I agree we should add more error handling and this is what we are planning to work on for the next PR which is more focus on the backend implementation and validation of reconciliation form.